cdc/sink/producer/kafka/config.go

```diff
@@ -0,0 +1,297 @@
+type Config struct {
+	BrokerEndpoints []string
+	PartitionNum    int32
+
+	// User should make sure that `replication-factor` not greater than the number of kafka brokers.
+	ReplicationFactor int16
+
+	Version         string
+	MaxMessageBytes int
+	Compression     string
+	ClientID        string
+	Credential      *security.Credential
+	SaslScram       *security.SaslScram
+	// control whether to create topic
+	AutoCreate bool
+}
+
+// NewConfig returns a default Kafka configuration
+func NewConfig() *Config {
+	return &Config{
+		Version: "2.4.0",
+		// MaxMessageBytes will be used to initialize producer, we set the default value (1M) identical to kafka broker.
+		MaxMessageBytes:   1 * 1024 * 1024,
+		ReplicationFactor: 1,
+		Compression:       "none",
+		Credential:        &security.Credential{},
+		SaslScram:         &security.SaslScram{},
+		AutoCreate:        true,
+	}
+}
+
+// CompleteByOpts the kafka configuration.
+func (c *Config) CompleteByOpts(sinkURI *url.URL, replicaConfig *config.ReplicaConfig, opts map[string]string) error {
+	c.BrokerEndpoints = strings.Split(sinkURI.Host, ",")
+	params := sinkURI.Query()
+	s := params.Get("partition-num")
+	if s != "" {
+		a, err := strconv.Atoi(s)
+		if err != nil {
+			return err
+		}
+		c.PartitionNum = int32(a)
+		if c.PartitionNum <= 0 {
+			return cerror.ErrKafkaInvalidPartitionNum.GenWithStackByArgs(c.PartitionNum)
+		}
+	}
+
+	s = params.Get("replication-factor")
+	if s != "" {
+		a, err := strconv.Atoi(s)
+		if err != nil {
+			return err
+		}
+		c.ReplicationFactor = int16(a)
+	}
+
+	s = params.Get("kafka-version")
+	if s != "" {
+		c.Version = s
+	}
+
+	s = params.Get("max-message-bytes")
+	if s != "" {
+		a, err := strconv.Atoi(s)
+		if err != nil {
+			return err
+		}
+		c.MaxMessageBytes = a
+		opts["max-message-bytes"] = s
+	}
+
+	s = params.Get("max-batch-size")
+	if s != "" {
+		opts["max-batch-size"] = s
+	}
+
+	s = params.Get("compression")
+	if s != "" {
+		c.Compression = s
+	}
+
+	c.ClientID = params.Get("kafka-client-id")
+
+	s = params.Get("ca")
+	if s != "" {
+		c.Credential.CAPath = s
+	}
+
+	s = params.Get("cert")
+	if s != "" {
+		c.Credential.CertPath = s
+	}
+
+	s = params.Get("key")
+	if s != "" {
+		c.Credential.KeyPath = s
+	}
+
+	s = params.Get("sasl-user")
+	if s != "" {
+		c.SaslScram.SaslUser = s
+	}
+
+	s = params.Get("sasl-password")
+	if s != "" {
+		c.SaslScram.SaslPassword = s
+	}
+
+	s = params.Get("sasl-mechanism")
+	if s != "" {
+		c.SaslScram.SaslMechanism = s
+	}
+
+	s = params.Get("auto-create-topic")
+	if s != "" {
+		autoCreate, err := strconv.ParseBool(s)
+		if err != nil {
+			return err
+		}
+		c.AutoCreate = autoCreate
+	}
+
+	s = params.Get("protocol")
+	if s != "" {
+		replicaConfig.Sink.Protocol = s
+	}
+
+	s = params.Get("enable-tidb-extension")
+	if s != "" {
+		_, err := strconv.ParseBool(s)
+		if err != nil {
+			return err
+		}
+		if replicaConfig.Sink.Protocol != "canal-json" {
```

Comment:
Should we check the boolean value of `enable-tidb-extension` is true here?

Reply:
It shouldn't be necessary, it should only be possible to configure this value in this protocol.

Reply:
I mean user configure sink-uri with `enable-tidb-extension=false&protocol=canal` or other protocols except for `canal-json`. Then the check will fail here.

Reply:
althrough `enable-tidb-extension=false` is useless.